aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/HttpResponseChecksumDecorator.kt

```diff
@@ -86,24 +129,56 @@ class HttpResponseChecksumCustomization(
+                                .map { algorithm -> algorithm.lowercase() }
+                                .joinToString(", ") { algorithm -> algorithm.dq() }
                         val runtimeApi = RuntimeType.smithyRuntimeApiClient(codegenContext.runtimeConfig)
                         rustTemplate(
                             """
                             #{ResponseChecksumInterceptor}::new(
                                 [$responseAlgorithms].as_slice(),
                                 |input: &#{Input}| {
-                                    ${""/* Per [the spec](https://smithy.io/2.0/aws/aws-core.html#http-response-checksums),
-                                           we check to see if it's the `ENABLED` variant */}
+                                    ${
+                                ""/* Per [the spec](https://smithy.io/2.0/aws/aws-core.html#http-response-checksums),
+                                           we check to see if it's the `ENABLED` variant */
+                            }
                                     let input: &#{OperationInput} = input.downcast_ref().expect("correct type");
                                     matches!(input.$validationModeName(), #{Some}(#{ValidationModeShape}::Enabled))
+                                },
+                                |input: &mut #{Input}, cfg: &#{ConfigBag}|  {
+                                    let input = input
+                                        .downcast_mut::<#{OperationInputType}>()
+                                        .ok_or("failed to downcast to #{OperationInputType}")?;
+
+                                    let request_validation_enabled =
+                                        matches!(input.$requestValidationModeName(), Some(#{ValidationModeShape}::Enabled));
+
+                                    if !request_validation_enabled {
+                                        // This value is set by the user on the SdkConfig to indicate their preference
+                                        let response_checksum_validation = cfg
+                                            .load::<#{ResponseChecksumValidation}>()
+                                            .unwrap_or(&#{ResponseChecksumValidation}::WhenSupported);
+
+                                        // If validation setting is WhenSupported (or unknown) we enable response checksum
+                                        // validation. If it is WhenRequired we do not enable (since there is no way to
+                                        // indicate that a response checksum is required).
+                                        ##[allow(clippy::wildcard_in_or_patterns)]
+                                        match response_checksum_validation {
+                                            #{ResponseChecksumValidation}::WhenRequired => {}
+                                            #{ResponseChecksumValidation}::WhenSupported | _ => {
```

Comment:
The `{Response | Request}ChecksumValidation` enums are marked `non-exhaustive` as a safety measure so have to handle that case somehow. Since these should always be set before this point that case shouldn't actually be encountered, but if it is we treat it the same as the default (`WhenSupported`).